trialfun/ft_trialfun_brainvision_segmented.m

```diff
@@ -97,18 +97,29 @@
 
 % add the stimulus events to the output, if possible
 numstim = cellfun(@length, stim);
-if all(numstim==numstim(1))
-  for i=1:length(stim)
-    for j=1:numstim(i)
-      stimvalue  = sscanf(value{stim{i}(j)}, cfg.trigformat);
-      stimsample = sample(stim{i}(j));
-      stimtime   = (stimsample - begsample(i) + offset(i))/hdr.Fs; % relative to 'Time 0'
-      trialinfo(i,2*(j-1)+1) = stimvalue;
-      trialinfo(i,2*(j-1)+2) = stimtime;
-    end % j
-  end % i
+if (numstim > 0)
+  if all(numstim==numstim(1))
+    for i=1:length(stim)
+      for j=1:numstim(i)
+        if isempty(value{stim{i}(j)})
+          error('missing a stimulus type definition in the related *.vmrk file');
+        end
+        stimvalue  = sscanf(value{stim{i}(j)}, cfg.trigformat);
+        stimsample = sample(stim{i}(j));
+        stimtime   = (stimsample - begsample(i) + offset(i))/hdr.Fs; % relative to 'Time 0'
+        if isempty(stimvalue)
+          error('the upper case letter of the stimulus value does not match with definition of "cfg.trigformat"'); 
```

Comment:
thanks for the suggestion. Can you please add the new option to the documentation (help) of the function, otherwise people won't know it exists. Furthermore, I would opt for calling the option "stimformat" rather than "trig format", since the other code and user-feedback also mentions stim/stimulus.